Maxim targets don't compile with blinky-baremetal

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

Few Maxim targets fail to compile blinky-baremetal in Mbed OS 6

```
Link: mbed-os-example-blinky-baremetal
[Warning] @0,0: L3912W: Option 'legacyalign' is deprecated.
[Error] @0,0: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Base (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
[Error] @0,0: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Length (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Base (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
Error: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Length (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
Finished: 0 information, 1 warning and 2 error messages.
[ERROR] Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Base (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
Error: L6218E: Undefined symbol Image$$ARM_LIB_HEAP$$ZI$$Length (referred from BUILD/MAX32600MBED/ARM/mbed-os/platform/source/mbed_sdk_boot.o).
Finished: 0 information, 1 warning and 2 error messages.
```

#### Target(s) affected by this defect ?

- MAX32600MBED
- MAX32620FTHR
- MAX32625MBED
- MAX32625PICO
- MAX32630FTHR

GCC and ARM Errors

#### Toolchain(s) (name and version) displaying this defect ?

Arm Compiler 6 and GCC 9

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-6.0.0

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Mbed CLI 1.10.4

#### How is this defect reproduced ? 

```
mbed import https://github.com/ARMmbed/mbed-os-example-blinky-baremetal
cd mbed-os-example-blinky
mbed compile -t <toolchain> -m <target>
```

@ARMmbed/team-maximintegrated

--- a/targets/TARGET_Maxim/TARGET_MAX32620C/device/TOOLCHAIN_ARM_STD/MAX32620.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32620C/device/TOOLCHAIN_ARM_STD/MAX32620.sct
@@ -4,8 +4,22 @@
 ; 256KB RAM (0x40000) @ 0x20000000
 
 
+#if !defined(MBED_APP_START)
+  #define MBED_APP_START  0x000000000
+#endif
+
+#if !defined(MBED_APP_SIZE)
+  #define MBED_APP_SIZE  0x200000
+#endif
 
 ; MAX32620: 2MB FLASH (0x200000) + 256KB RAM (0x40000)
+#if !defined(MBED_RAM_SIZE)
+#define MBED_RAM_SIZE 0x40000
+#endif
+
+#if !defined(MBED_RAM_START)
+#define MBED_RAM_START 0x20000000
+#endif
 
 #if !defined(MBED_CONF_TARGET_BOOT_STACK_SIZE)
 #  if defined(MBED_BOOT_STACK_SIZE)
@@ -15,20 +29,23 @@
 #  endif
 #endif
 
+; [RAM] Vector table dynamic copy: 65 vectors * 4 bytes = 260 (0x104) + 4
+; for 8 byte alignment
+#define VECTOR_SIZE 0x108
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
-LR_IROM1 0x00000000 0x200000  {    ; load region size_region
-    ER_IROM1 0x00000000 0x200000  {  ; load address = execution address 
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
+    ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)
     }
-
-    ; [RAM] Vector table dynamic copy: 65 vectors * 4 bytes = 260 (0x104) + 4
-    ; for 8 byte alignment
-    RW_IRAM1 (0x20000000+0x108) (0x40000-0x108-Stack_Size)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_STACK (0x20000000+0x40000) EMPTY -Stack_Size { ; stack
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    }
+    ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }
 }

--- a/targets/TARGET_Maxim/TARGET_MAX32620C/device/TOOLCHAIN_ARM_STD/MAX32620.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32620C/device/TOOLCHAIN_ARM_STD/MAX32620.sct
@@ -35,16 +35,19 @@
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
 #define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
+#define MBED_RAM1_START (MBED_RAM_START+VECTOR_SIZE)
+#define MBED_RAM1_SIZE (MBED_RAM_SIZE-RAM_FIXED_SIZE)
+
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)
     }
-    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
+    RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)){ ; heap region growing up
     }
     ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }

--- a/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_BOOT/MAX32625.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_BOOT/MAX32625.sct
@@ -32,6 +32,9 @@
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
 #define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
+#define MBED_RAM1_START (MBED_RAM_START+VECTOR_SIZE)
+#define MBED_RAM1_SIZE (MBED_RAM_SIZE-RAM_FIXED_SIZE)
+
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address
         *.o (RESET, +First)
@@ -39,11 +42,11 @@ LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
         .ANY (+RO)
     }
 
-    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
+    RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE  {  ; RW data
         .ANY (+RW +ZI)
     }
 
-    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)){ ; heap region growing up
     }
     ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }

--- a/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_BOOT/MAX32625.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_BOOT/MAX32625.sct
@@ -11,6 +11,14 @@
     #define MBED_APP_SIZE 0x00070000
 #endif
 
+#if !defined(MBED_RAM_SIZE)
+#define MBED_RAM_SIZE 0x28000
+#endif
+
+#if !defined(MBED_RAM_START)
+#define MBED_RAM_START 0x20000000
+#endif
+
 #if !defined(MBED_CONF_TARGET_BOOT_STACK_SIZE)
 #  if defined(MBED_BOOT_STACK_SIZE)
 #    define MBED_CONF_TARGET_BOOT_STACK_SIZE MBED_BOOT_STACK_SIZE
@@ -19,7 +27,10 @@
 #  endif
 #endif
 
+; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
+#define VECTOR_SIZE 0x110
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address
@@ -28,10 +39,12 @@ LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
         .ANY (+RO)
     }
 
-    ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (0x20000000+0x110) (0x28000-0x110-Stack_Size)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_STACK (0x20000000+0x28000) EMPTY -Stack_Size { ; stack
+
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    }
+    ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }
 }

--- a/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_NO_BOOT/MAX32625.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_NO_BOOT/MAX32625.sct
@@ -3,6 +3,22 @@
 ; 512KB FLASH (0x80000) @ 0x000000000
 ; 160KB RAM (0x28000) @ 0x20000000
 
+#if !defined(MBED_APP_START)
+    #define MBED_APP_START 0x000000000
+#endif
+
+#if !defined(MBED_APP_SIZE)
+    #define MBED_APP_SIZE 0x80000
+#endif
+
+#if !defined(MBED_RAM_SIZE)
+#define MBED_RAM_SIZE 0x28000
+#endif
+
+#if !defined(MBED_RAM_START)
+#define MBED_RAM_START 0x20000000
+#endif
+
 #if !defined(MBED_CONF_TARGET_BOOT_STACK_SIZE)
 #  if defined(MBED_BOOT_STACK_SIZE)
 #    define MBED_CONF_TARGET_BOOT_STACK_SIZE MBED_BOOT_STACK_SIZE
@@ -11,19 +27,24 @@
 #  endif
 #endif
 
+; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
+#define VECTOR_SIZE 0x110
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
-LR_IROM1 0x000000000 0x80000  {    ; load region size_region
-    ER_IROM1 0x000000000 0x80000  {  ; load address = execution address 
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
+    ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (0x20000000+0x110) (0x28000-0x110-Stack_Size)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_STACK (0x20000000+0x28000) EMPTY -Stack_Size { ; stack
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    }
+    ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }
 }

--- a/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_NO_BOOT/MAX32625.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32625/device/TOOLCHAIN_ARM_STD/TARGET_MAX32625_NO_BOOT/MAX32625.sct
@@ -32,6 +32,9 @@
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
 #define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
+#define MBED_RAM1_START (MBED_RAM_START+VECTOR_SIZE)
+#define MBED_RAM1_SIZE (MBED_RAM_SIZE-RAM_FIXED_SIZE)
+
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
@@ -40,10 +43,10 @@ LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
+    RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)){ ; heap region growing up
     }
     ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }

--- a/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
@@ -34,7 +34,7 @@
 #define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
-    ER_IROM1 MBED_APP_START MBED_APP_START  {  ; load address = execution address 
+    ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)

--- a/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
@@ -3,6 +3,23 @@
 ; 2MB FLASH (0x200000) @ 0x000000000
 ; 512KB RAM (0x80000) @ 0x20000000
 
+#if !defined(MBED_APP_START)
+  #define MBED_APP_START  0x000000000
+#endif
+
+#if !defined(MBED_APP_SIZE)
+  #define MBED_APP_SIZE  0x200000
+#endif
+
+; MAX32620: 2MB FLASH (0x200000) + 256KB RAM (0x40000)
+#if !defined(MBED_RAM_SIZE)
+#define MBED_RAM_SIZE 0x80000
+#endif
+
+#if !defined(MBED_RAM_START)
+#define MBED_RAM_START 0x20000000
+#endif
+
 #if !defined(MBED_CONF_TARGET_BOOT_STACK_SIZE)
 #  if defined(MBED_BOOT_STACK_SIZE)
 #    define MBED_CONF_TARGET_BOOT_STACK_SIZE MBED_BOOT_STACK_SIZE
@@ -11,19 +28,24 @@
 #  endif
 #endif
 
+; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
+#define VECTOR_SIZE 0x110
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
-LR_IROM1 0x000000000 0x200000  {    ; load region size_region
-    ER_IROM1 0x000000000 0x200000  {  ; load address = execution address 
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
+    ER_IROM1 MBED_APP_START MBED_APP_START  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (0x20000000+0x110) (0x80000-0x110-Stack_Size)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_START-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_STACK (0x20000000+0x80000) EMPTY -Stack_Size { ; stack
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    }
+    ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }
 }

--- a/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
@@ -33,6 +33,9 @@
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
 #define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
+#define MBED_RAM1_START (MBED_RAM_START+VECTOR_SIZE)
+#define MBED_RAM1_SIZE (MBED_RAM_SIZE-RAM_FIXED_SIZE)
+
 LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
@@ -41,10 +44,10 @@ LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
+    RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)){ ; heap region growing up
     }
     ARM_LIB_STACK (MBED_RAM_START+MBED_RAM_SIZE) EMPTY -Stack_Size { ; stack region growing down
     }

--- a/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
+++ b/targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct
@@ -41,7 +41,7 @@ LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_START-RAM_FIXED_SIZE)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
     ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up

--- a/targets/targets.json
+++ b/targets/targets.json
@@ -5184,7 +5184,6 @@
         ],
         "supported_toolchains": [
             "GCC_ARM",
-            "IAR",
             "ARM"
         ],
         "device_has": [
@@ -5206,7 +5205,18 @@
         ],
         "release_versions": [
             "5"
-        ]
+        ],
+        "supported_application_profiles" : ["full", "bare-metal"],
+        "supported_c_libs": {
+            "arm": [
+                "std",
+                "small"
+            ],
+            "gcc_arm": [
+                "std",
+                "small"
+            ]
+        }
     },
     "MAX32625_BASE": {
         "inherits": [

